great_expectations/data_context/config_validator/yaml_config_validator.py

```diff
@@ -0,0 +1,623 @@
+                yaml_config,
+                substitutions,
+            )
+        except Exception as e:
+            usage_stats_event_payload: dict = {
+                "diagnostic_info": ["__substitution_error__"],
+            }
+            send_usage_message_from_handler(
+                event=usage_stats_event_name,
+                handler=self.usage_statistics_handler,
+                event_payload=usage_stats_event_payload,
+                success=False,
+            )
+            raise e
+
+        try:
+            config: CommentedMap = yaml.load(config_str_with_substituted_variables)  # type: ignore[arg-type]
+            return config
+
+        except Exception as e:
+            usage_stats_event_payload = {
+                "diagnostic_info": ["__yaml_parse_error__"],
+            }
+            send_usage_message_from_handler(
+                event=usage_stats_event_name,
+                handler=self.usage_statistics_handler,
+                event_payload=usage_stats_event_payload,
+                success=False,
+            )
+            raise e
+
+    def _test_instantiation_of_store_from_yaml_config(
+        self, name: Optional[str], class_name: str, config: CommentedMap
+    ) -> Tuple[Store, dict]:
+        """
+        Helper to create store instance and update usage stats payload.
+        See `test_yaml_config` for more details.
+        """
+        print(f"\tInstantiating as a Store, since class_name is {class_name}")
+        store_name: str = name or config.get("name") or "my_temp_store"
+        instantiated_class = cast(
+            Store,
+            self._data_context._build_store_from_config(
+                store_name=store_name,
+                store_config=config,
+            ),
+        )
+        store_name = instantiated_class.store_name or store_name
+        self._data_context.config["stores"][store_name] = config  # type: ignore[index]
+
+        anonymizer = Anonymizer(self._data_context.data_context_id)
+        usage_stats_event_payload = anonymizer.anonymize(
+            store_name=store_name, store_obj=instantiated_class  # type: ignore[arg-type]
+        )
+        return instantiated_class, usage_stats_event_payload
+
+    def _test_instantiation_of_datasource_from_yaml_config(
+        self, name: Optional[str], class_name: str, config: CommentedMap
+    ) -> Tuple[Datasource, dict]:
+        """
+        Helper to create datasource instance and update usage stats payload.
+        See `test_yaml_config` for more details.
+        """
+        print(f"\tInstantiating as a Datasource, since class_name is {class_name}")
+        datasource_name: str = name or config.get("name") or "my_temp_datasource"
+        datasource_config = datasourceConfigSchema.load(config)
+        datasource_config.name = datasource_name
+        instantiated_class = cast(
+            Datasource,
+            self._data_context._instantiate_datasource_from_config_and_update_project_config(
```

Comment:
I have reservations about the access of private methods here but understand why we're doing it (was previously not an issue when this method was within the same class).